html/php-challenge/index.php

```diff
@@ -74,61 +74,139 @@ function makeLink($value) {
+			<input type="hidden" name="reply_post_id" value="<?php echo h($_REQUEST['res']); ?>" />
+			</dd>
+		</dl>
+		<div>
+			<p>
+			<input type="submit" value="投稿する" />
+			</p>
+		</div>
+		</form>
+
+<?php foreach ($posts as $post): ?>
     <div class="msg">
-    <img src="member_picture/<?php echo h($post['picture']); ?>" width="48" height="48" alt="<?php echo h($post['name']); ?>" />
-    <p><?php echo makeLink(h($post['message'])); ?><span class="name">（<?php echo h($post['name']); ?>）</span>[<a href="index.php?res=<?php echo h($post['id']); ?>">Re</a>]</p>
-    <p class="day"><a href="view.php?id=<?php echo h($post['id']); ?>"><?php echo h($post['created']); ?></a>
-		<?php
-if ($post['reply_post_id'] > 0):
-?>
-<a href="view.php?id=<?php echo
-h($post['reply_post_id']); ?>">
-返信元のメッセージ</a>
-<?php
-endif;
-?>
-<?php
-if ($_SESSION['id'] == $post['member_id']):
-?>
-[<a href="delete.php?id=<?php echo h($post['id']); ?>"
-style="color: #F33;">削除</a>]
-<?php
-endif;
-?>
-    </p>
+		<!-- 誰がリツイートしたかリツイート、表示 start ====================================-->
+			<?php if ($post['originally_id'] > 0 && $post['member_id'] == $_SESSION['id']): ?>
+				<span style='font-size: 12px; color: #c0c0c0;'>リツイート済み</span>
+			<?php elseif ($post['originally_id'] > 0 && $post['member_id'] != $_SESSION['id']): ?>
+				<span style='font-size: 12px; color: #c0c0c0;'><?php echo h($post['name']); ?>さんがリツイート</span>
+			<?php endif; ?>
+		<!-- 誰がリツイートしたかリツイート、表示　end ====================================-->
+
+		<img src="member_picture/<?php echo h($post['picture']); ?>" width="48" height="48" style="border-radius:50%;" alt="<?php echo h($post['name']); ?>" />
+		<p><?php echo makeLink(h($post['message'])); ?><span class="name">（<?php echo h($post['name']); ?>）</span>[<a href="index.php?res=<?php echo h($post['id']); ?>">Re</a>]</p>
+		<p class="day">
+			<a href="view.php?id=<?php echo h($post['id']); ?>"><?php echo h($post['created']); ?></a>
+
+		<!-- リツイート start  -->
+			<!-- リツイートの色判定 start ====================================-->
+			<?php
+				// retweets_countテーブルの中からmessageが一緒のやつをとる
+				$retweets_count_table = $db->prepare('SELECT * FROM retweets_count WHERE post_message=?');
+				$retweets_count_table->bindParam(1,$post['message']);
+				$retweets_count_table->execute();
+				$retweets_count_tables = $retweets_count_table->fetchAll(PDO::FETCH_ASSOC);
+
+				//retweets_countテーブルのレコード数を数えてくれる
+				$retweet_count_table_sql = 'SELECT COUNT(*) FROM retweets_count';
+				$stmt = $db->query($retweet_count_table_sql);
+				$retweet_count_table_amount = (int)$stmt->fetchColumn();
+				
+				for($i=0;$i<$retweet_count_table_amount;$i++){
+					$is_retweet = '';
+					if($retweets_count_tables[$i]['post_message'] == $post['message'] && !empty($retweets_count_tables[$i]['member_id'] == $_SESSION['id'])){
+						$is_retweet = 'true';
+						break;
+					}else{
+						$is_retweet = '';
+					}
+				}
+			?>
+			<?php if($is_retweet == 'true'): ?>
```

Comment:
$is_retweetをbooleanにして、こう書いた方がよいです
```suggestion
			<?php if($is_retweet): ?>
```